app/models/katello/root_repository.rb

```diff
@@ -333,6 +334,28 @@ def format_arches
       end
     end
 
+    def update_audit_password
+      audit = audits.last
+
+      if audit.audited_changes.key? "upstream_password"
+        encrypt_password_for_audit(audit)
+      end
+    end
```

Comment:
is audits.last really guaranteed to return the previously saved audit entry? 

You might could order by :created_at to try to ensure that, but i feel like it would be better to extend the audit model and add a before_save callback to modify the audited_changes hash in a similar way that you've done here (if the associated model is a RootRepository)

Reply:
here's where we 'extend' other models/classes: https://github.com/Katello/katello/blob/master/lib/katello/engine.rb#L155-L170

Reply:
might wanna explore https://github.com/theforeman/foreman/blob/develop/app/models/lookup_keys/lookup_key.rb#L2